Algorithm.CSharp/ForexMultiResolutionRegressionAlgorithm.cs

```diff
@@ -0,0 +1,129 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using System;
+using System.Linq;
+using QuantConnect.Data;
+using QuantConnect.Interfaces;
+using System.Collections.Generic;
+
+namespace QuantConnect.Algorithm.CSharp
+{
+    /// <summary>
+    /// This algorithm is a test case for forex symbols at multiple resolutions.
+    /// </summary>
+    public class ForexMultiResolutionRegressionAlgorithm : QCAlgorithm, IRegressionAlgorithmDefinition
+    {
+        private readonly Dictionary<Symbol, int> _dataPointsPerSymbol = new();
+
+        /// <summary>
+        /// Initialise the data and resolution required, as well as the cash and start-end dates for your algorithm. All algorithms must initialized.
+        /// </summary>
+        public override void Initialize()
+        {
+            SetStartDate(2013, 10, 7);
+            SetEndDate(2013, 10, 8);
+
+            var eurgbp = AddForex("EURGBP", Resolution.Daily);
+            _dataPointsPerSymbol.Add(eurgbp.Symbol, 0);
+
+            var gbpusd = AddForex("EURUSD", Resolution.Hour);
+            _dataPointsPerSymbol.Add(gbpusd.Symbol, 0);
+        }
+
+        /// <summary>
+        /// OnData event is the primary entry point for your algorithm. Each new data point will be pumped in here.
+        /// </summary>
+        /// <param name="data">Slice object keyed by symbol containing the stock data</param>
+        public override void OnData(Slice data)
+        {
+            foreach (var kvp in data)
+            {
+                var symbol = kvp.Key;
+                _dataPointsPerSymbol[symbol]++;
+
+                if (symbol == "EURUSD" && kvp.Value.IsFillForward)
+                {
+                    throw new Exception($"Unexpected fill forward for 'EURUSD' bar at {Time}");
+                }
+            }
+        }
+
+        /// <summary>
+        /// End of algorithm run event handler. This method is called at the end of a backtest or live trading operation. Intended for closing out logs.
+        /// </summary>
+        public override void OnEndOfAlgorithm()
+        {
+            if (// first bat at '10/7/2013 8:00:00 PM' + hour FFing until next bar '10/8/2013 8:00:00 PM' + 4 more FF bars until 10/9/2013 12:00:00 AM
+                _dataPointsPerSymbol["EURGBP"] != 29
+                // data from '10/7/2013 12:00:00 AM' to '10/9/2013 12:00:00 AM'
+                || _dataPointsPerSymbol["EURUSD"] != 49)
```

Comment:
Not a big deal, but I'd expect "SetEndDate(2013, 10, 8);" to mean midnight on the 8th / an implicit 0 timespan for the 8th. So this would be 24 hours -> 24/25 bars, not 49.